net/datacenter-node-install/setup-dc-node-1.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+if [[ "$EUID" -ne 0 ]]; then
+  echo "Please run as root"
+  exit 1
+fi
+
+if [[ -n "$1" ]]; then
+  PUBKEY_FILE="$1"
+else
+  cat <<EOF
+Usage: $0 [pubkey_file]
+
+The pubkey_file should be the pubkey that will eb set up to allow the current user
+(assumed to be the machine admin) to log in via ssh
+EOF
+  exit 1
+fi
+
+USERNAME="$(whoami)"
+HERE="$(dirname "$0")"
+
+set -xe
+
+apt update
+apt upgrade -y
+apt install -y build-essential pkg-config clang
+
+"$HERE"/../scripts/install-docker.sh
+usermod -aG docker "$USERNAME"
+"$HERE"/../scripts/install-certbot.sh
+"$HERE"/setup-sudoers.sh
+"$HERE"/setup-ssh.sh
+
+# Allow admin user to log in
+BASE_SSH_DIR="${HOME}/.ssh"
+mkdir "$BASE_SSH_DIR"
```

Comment:
Won't $HOME be root's homedir in this case?  Looks like you want $USERNAME's homedir, no?
Can use
`su - ${USERNAME} -c 'mkdir "$BASE_SSH_DIR"'` and others, so you don't need to `chown` after every file/dir creation

Reply:
I just did the obvious thing to communicate the intent of the change.  There are bigger problems, see https://github.com/solana-labs/solana/pull/5517#discussion_r315990902